src/spaces/discrete.jl

```diff
@@ -117,6 +117,25 @@ function random_empty(model::ABM{<:DiscreteSpace}, cutoff = 0.998)
     end
 end
 
+"""
+    empty_nearby_positions(position, model::ABM{<:DiscreteSpace}, r = 1; kwargs)
+    empty_nearby_positions(agent, model::ABM{<:DiscreteSpace}, r = 1; kwargs)
+
+Return an iterable of all empty positions within "radius" `r` of the given `position` or `agent`.
+
+The value of r and possible keywords operate identically to `nearby_positions`.
+
+This function only exists for discrete spaces with a finite amount of positions.
+"""
+
+function empty_nearby_positions(agent::A, model, r = 1) where {A<:AbstractAgent}
+    return empty_nearby_positions(agent.pos, model, r)
+end
+function empty_nearby_positions(pos, model, r = 1)
+    return Iterators.filter(x -> isempty(x, model), nearby_positions(pos, model, r))
```

Comment:
```suggestion
    return Iterators.filter(Base.Fix2(isempty, model), nearby_positions(pos, model, r))
```
See https://github.com/SciML/SciMLStyle#closures-should-be-avoided-whenever-possible